support/tarball/tool-bootstrapping/bootstrap.sh

```diff
@@ -0,0 +1,231 @@
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ ! -d "$stage1Dir" ]; then
+        echo "$stage1Dir doesn't exist"
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ ! -d "$refPkgsDir" ]; then
+        echo "$refPkgsDir doesn't exist"
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ ! -d "$tarballSourceDir" ]; then
+        echo "$tarballSourceDir doesn't exist"
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ "${1-default}" != "test" ]; then
+    LogMessage "Copy tarball-source to ./final-sdk"
+    cp -r $tarballSourceDir $finalSdkDir
+    fi
+
+    LogMessage "Remove archive packages, source-built packages and SDK from final-sdk dir"
+    rm -rf $finalSdkDir/packages/archive/*
+    rm -rf $finalSdkDir/packages/source-built/*
+    rm -rf $finalSdkDir/.dotnet
+
+    LogMessage "Copy stage1 coreclr-tools to final-sdk"
+    cp -r $stage1Dir/Tools/source-built/coreclr-tools $finalSdkDir/packages/source-built
+
+    LogMessage "Copy source-built packages from stage1 build output to final-sdk packages dir"
+    cp $stage1Dir/bin/x64/Release/Private.SourceBuilt.Artifacts*.tar.gz $finalSdkDir/packages/archive
+
+    LogMessage "Copy .NET Core SDK from stage1 build output to final-sdk"
+    if [ -d "$finalSdkDir/.dotnet/" ]; then
+        rm -rf $finalSdkDir/.dotnet/
+    fi
+    mkdir $finalSdkDir/.dotnet/
+    tar -xzf $stage1Dir/bin/x64/Release/dotnet-sdk-3*.tar.gz --directory $finalSdkDir/.dotnet/
+
+    LogMessage "Copy reference packages from reference-packages to final-sdk"
+    cp $refPkgsDir/artifacts/Private.SourceBuild.ReferencePackages.bootstrap.tar.gz $finalSdkDir/packages/archive
+
+    LogMessage "Update version of SDK and arcade in global.json"
+    arcadeName=$(tar -tf $finalSdkDir/packages/archive/Private.SourceBuilt.Artifacts*.tar.gz Microsoft.DotNet.Arcade.Sdk.*.nupkg)
+    arcadeVer=${arcadeName//Microsoft.DotNet.Arcade.Sdk.}
+    arcadeVer=${arcadeVer//.nupkg}
+    LogMessage "    Arcade version = [$arcadeVer]"
+    sed -i "s|\(\"Microsoft.DotNet.Arcade.Sdk\": \"\)[^\"]*\"|\1$arcadeVer\"|g" $finalSdkDir/global.json
+
+    sdkVer=$(ls $finalSdkDir/.dotnet/sdk)
+    LogMessage "    Sdk version = [$sdkVer]"
+    sed -i "s|\"dotnet\": \".*|\"dotnet\": \"$sdkVer\"|g" $finalSdkDir/global.json
+
+    LogMessage "Build final sdk"
+    cd $finalSdkDir
+    ./build.sh
+
+    LogMessage "Completing Step3 - Build final sdk"
+}
+
+if [ -z "${3:-}" ]; then
+    usage
+    exit 1
+fi
+
+bootstrapDir=$1
+tarballSourceDir=$2
+refPkgSourceDir=$3
+startStep="${4:-}"
+
+stage1Dir="$bootstrapDir/stage1-sdk"
+refPkgsDir="$bootstrapDir/reference-packages"
+finalSdkDir="$bootstrapDir/final-sdk"
+
+if [[ -d "$bootstrapDir" && "$startStep" == "" ]]; then
+    echo "$bootstrapDir already exists"
+    echo "exiting..."
+    exit 1
+fi
+
+if [ ! -d "$bootstrapDir" ]; then
+    mkdir "$bootstrapDir"
+fi
+
+# Check network access
+wget -q --spider http://www.microsoft.com
```

Comment:
I know this is just a quick check but do we want to check that wget exists before relying on it?  I think several of the dotnet-buildtools-prereqs images don't include it.

Reply:
I'll do that.  Thanks.